boards/xtensa/esp32/doc/index.rst

```diff
@@ -48,30 +48,48 @@ System requirements
 Prerequisites
 =============
 
-Two components are required in order to build this port: the `toolchain`_
-and the `SDK`_.  Both are provided by the SoC manufacturer.
+Some tools need to be installed on the computer before proceeding to the next steps.
+Follow the `installing prerequisites`_ instructions for your OS.
 
-The SDK contains headers and a hardware abstraction layer library
-(provided only as object files) that are required for the port to
-function.
+In order to build this port, the `SDK`_ and the `toolchain`_ need
+to be manually addressed. Both are provided by the SoC manufacturer.
+Cloning ESP_IDF is not necessary as it is automatically fetched in external modules.
```

Comment:
From what I can tell, this would install in the location recommended in the upstream documentation, because `west expressif update` for me did a few things then said the toolchains were already installed.  If so, please document that: some of us would get upset when things are silently installed in `${HOME}`.

Because that installation done behind the scenes with the west command it might be helpful to change the text below that's currently:

    export ESPRESSIF_TOOLCHAIN_PATH="/path/to/xtensa-esp32-elf/"

to give a better hint of what it needs to be provided.  In my case it's probably:

    ${HOME}/.espressif/tools/xtensa-esp32-elf/esp-2020r3-8.4.0/xtensa-esp32-elf/

which is a mouthful.  It also looks like that would have to change for s2 and s3 as those seem to have different toolchains accessed through a different mid-level path component.

Reply:
@pabigot, If I understand:
1) Write down on the getting started tutorial what `west espressif update` will perform in details;
2) Change the export statement regarding xtensa toolchain;

As we have no support yet for S2 and S3, I think we can keep `(...)/xtensa-esp32-elf/` as is. Ok?

Reply:
Sure.  But if it's true that supporting esp32s2 and esp32s3 would require changing the `EXPRESSIF_TOOLCHAIN_PATH` depending on the target it would be worth trying to figure out how that can be avoided.

It'd be nice to have one workspace environment that handles all ESP32 boards, just as we do now for ARM.  (RISCV tends to default to different toolchains per SOC, but those SOCs come from different vendors so that's a little easier to justify.)

But if we would have to switch, it's likely to be important to make it easier to do so.  One option could be to have the user source `${HOME}/.expressif/zephyr-esp32.sh` (`...-esp32s2.sh`, ...) to set `EXPRESSIF_TOOLCHAIN_PATH` correctly, where those setup scripts could be generated by `west expressif update`.

That can be deferred, though by doing it now you'd eliminate the need to get the installation-specific toolchain version element `esp-2020r3-8.4.0` into the environment variable.

Reply:
> 1. Write down on the getting started tutorial what `west espressif update` will perform in details;

For that you could just point to the upstream documentation, and say "`west espressif update` will execute steps 1, 2, 3, 4 (?) of (the upstream documentation)", with a *short* summary of what that means.  No need to go into details that are documented elsewhere.

Reply:
Regarding the toolchain path, we are working to improve that in a way no switching will be required among SoCs. However, this might take a few weeks. Can we keep current approach for now?

Reply:
Absolutely, yes.

Reply:
@pabigot, 
I have updated the getting started document and installation process. Let me know how that sounds.

- User can either download ESP32 toolchain manually or run `west` command to automatically retrieve it.

- Added the download path, export statement and notes so that user won't be in the dark with the installation environment.

Regarding other ESP32 SoC support, we will handle that in a another moment. For now, let's work with this so that pending ESP32 features can be merged as well.